### PR TITLE
Plugin version check test

### PR DESCRIPTION
**Pull request by hairyhum**
Test for https://github.com/rabbitmq/rabbitmq-server/issues/591.

### CODE REVIEW

File: test/src/rabbit_plugins_tests.erl [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     {[], "any version", ok} % any version
+    ,{[], "0.0.0", ok}       % any version
+    ,{[], "3.5.6", ok}       % any version
+    ,{["something"], "something", ok}      % Equal always match 
+    ,{["3.5.4"], "something", err}
+    ,{["something", "3.5.6"], "3.5.7", ok} % any version can match
+    ,{["3.4.0", "3.5.6"], "3.6.1", err}    % major version should present
+    ,{["3.5.2", "3.6.1"], "3.5.2", ok}     % equal version match
+    ,{["3.5.2", "3.6.1"], "3.5.1", err}    % lesser version don't match
+    ,{["3.5.2", "3.6.1"], "3.6.2", ok}     % greated version match
```
Comment (michaelklishin) [inline test/src/rabbit_plugins_tests.erl]: Why is `3.6.2` accepted here?

File: test/src/rabbit_plugins_tests.erl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+-compile(export_all).
+
+test_version_support() ->
+    Examples = [
+     {[], "any version", ok} % any version
+    ,{[], "0.0.0", ok}       % any version
+    ,{[], "3.5.6", ok}       % any version
+    ,{["something"], "something", ok}      % Equal always match 
+    ,{["3.5.4"], "something", err}
+    ,{["something", "3.5.6"], "3.5.7", ok} % any version can match
```
Comment (michaelklishin) [inline test/src/rabbit_plugins_tests.erl]: I don't get why `3.5.7` is accepted here?

File: test/src/rabbit_plugins_tests.erl (hunk 1 above)
Comment (hairyhum) [inline test/src/rabbit_plugins_tests.erl]: Because "3.6.1" is minimal version for "3.6" major

**Comment by hairyhum** [comment]
It's not range. It's list of minimal versions of major version branches.

**Comment by hairyhum** [comment]
Sorry for rewriting changes. Added one more branch to reduce confusion.

**Pull request merged**